pyccel/parser/semantic.py

```diff
@@ -3423,8 +3409,15 @@ def _visit_FunctionDef(self, expr):
 
                 # update the class methods
 
-                self.scope.insert_class(ClassDef(cls_name, cls.attributes,
-                        methods, superclass=cls.superclass))
+                superclasses = [self.scope.find(s, 'classes') for s in cls.superclasses]
+                if any(s is None for s in superclasses):
+                    for s in superclasses:
+                        if s is None:
+                            errors.report("Couldn't find class {s} in scope", symbol=expr,
+                                    severity='error')
+                    superclasses = [s for s in superclasses if s is not None]
```

Comment:
This block of code also needs to be fixed like the one in `_visit_ClassDef`

Reply:
Fixed duplication : 5c49581e